kratos/processes/calculate_nodal_area_process.h

```diff
@@ -59,38 +43,75 @@ typedef  ModelPart::ElementsContainerType ElementsContainerType;
 ///@name Kratos Classes
 ///@{
 
-/// Short class definition.
-/** Detail class definition.
-*/
-class CalculateNodalAreaProcess
+/**
+ * @brief This struct is used in order to identify when using the hitorical and non historical variables
+ */
+struct CalculateNodalAreaSettings
+{
+    // Defining clearer options
+    constexpr static bool SaveAsHistoricalVariable = true;
+    constexpr static bool SaveAsNonHistoricalVariable = false;
+};
+    
+/** 
+ * @class CalculateNodalAreaProcess
+ * @ingroup KratosCore 
+ * @brief Computes NODAL_AREA
+ * @details Calculate the NODAL_AREA for computing the weighted area in each node
+ * @author Riccardo Rossi
+ * @author Vicente Mataix Ferrandiz
+ */
+template<bool THistorical = true>
+class KRATOS_API(KRATOS_CORE) CalculateNodalAreaProcess
     : public Process
 {
 public:
     ///@name Type Definitions
     ///@{
 
+    /// Index type definition
+    typedef std::size_t IndexType;
+    
+    /// Size type definition
+    typedef std::size_t SizeType;
+
+    /// The definition of the node
+    typedef Node<3> NodeType;
+
+    /// The definition of the node iterator
+    typedef ModelPart::NodeIterator NodeIterator;
+    
     /// Pointer definition of CalculateNodalAreaProcess
     KRATOS_CLASS_POINTER_DEFINITION(CalculateNodalAreaProcess);
 
     ///@}
     ///@name Life Cycle
     ///@{
 
-    /// Default constructor.
-    /// avg_elems ------ expected number of neighbour elements per node.,
-    /// avg_nodes ------ expected number of neighbour Nodes
-    /// the better the guess for the quantities above the less memory occupied and the fastest the algorithm
-    CalculateNodalAreaProcess(ModelPart& model_part, unsigned int domain_size)
-        : mr_model_part(model_part), mdomain_size(domain_size)
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param DomainSize The size of the space, if the value is not provided will compute from the model part
+     */
+    CalculateNodalAreaProcess(
+        ModelPart& rModelPart, 
+        const SizeType DomainSize = 0
+        ): mrModelPart(rModelPart),
+           mDomainSize(DomainSize)
     {
+        // In case is not provided we will take from the model part
+        if (mDomainSize == 0) {
+            const auto& it_element_begin = mrModelPart.ElementsBegin();
+            const auto& r_first_element_geometry = it_element_begin->GetGeometry();
+            mDomainSize = r_first_element_geometry.WorkingSpaceDimension();
```

Comment:
Are you sure you want to give the user the option for automatizing this?
This is not done in the current version.

If you do, then why not getting it from the ProcessInfo?

Reply:
The current version only works with a very limited geometries (tri and tetra). This code is basically for retrocompatibility

Reply:
I know, but your current version again has a problem if there are different elems in a modelpart

Suggestion:
~~~c
CalculateNodalAreaProcess(
    ModelPart& rModelPart, 
    const SizeType DomainSize
    ): mrModelPart(rModelPart),
        mDomainSize(DomainSize) 
    {}

CalculateNodalAreaProcess(
    ModelPart& rModelPart
    ): mrModelPart(rModelPart)
    {
        KRATOS_ERROR_IF_NOT(rModelPart.GetProcessInfo().Has(DOMAIN_SIZE))
            << "\"DOMAIN_SIZE\" has to be specified in "
            << "the ProcessInfo if using this constructor!"
        mDomainSize = rModelPart.GetProcessInfo()[DOMAIN_SIZE]
    }
~~~

this way you are on the safe side in any case

Reply:
OK